### PR TITLE
Milestone2 - redux reducers store

### PR DESCRIPTION
**Pull request by maelfosso**
Hello,

Now, it's possible to 
- create a book
- remove a book

Please, could you tell me if everything is OK?
Thanks.

### CODE REVIEW

**Comment by maelfosso** [comment]
I just read the [Milestone 3](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896828/), and It like I've already implemented the feature asked.

How could I handle that, please?

File: src/App.js [hunk 1]
```diff
@@ -1,14 +1,13 @@
 import React from 'react';
+import BooksList from './components/BooksList';
+import BooksForm from './components/BooksForm';
 import './App.css';
 
 function App() {
```
Comment (bellom) [inline src/App.js:6]: Let add `App.js` inside the component folder created, as we both know it one of the presentational components.
image

File: src/components/Book.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      <td>{book.title}</td>
+      <td>{book.category}</td>
+      <td><button type="button" onClick={onRemove}>Remove</button></td>
+    </tr>
+  );
+};
+
+Book.propTypes = {
+  book: PropTypes.shape({
+    id: PropTypes.number.isRequired,
```
Comment (bellom) [inline src/components/Book.js:18]: We are setting propTypes `id` but we are not using it inside the code.

File: src/components/Book.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Book = props => {
+  const { book, onRemove } = props;
+  return (
+    <tr>
+      <td />
```
Comment (bellom) [inline src/components/Book.js:8]: we should also have `book.id` here, right?
image

File: src/components/BooksList.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
```
Comment (bellom) [inline src/components/BooksList.js:1]: `BookList.js` should be inside `container folder` cause is one the container component.
For any component connecting to the redux store, we classify them as container component [read more](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0)
image

File: src/components/BooksForm.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
```
Comment (bellom) [inline src/components/BooksForm.js:1]: `BooksForm.js` should be inside `container folder` cause is one the container component.
For any component connecting to the redux store, we classify them as container component [read more](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0)
image

File: src/components/BooksForm.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { createBook } from '../actions';
+import CATEGORIES from '../utils/categories';
+
+class BooksForm extends React.Component {
+  constructor(props) {
```
Comment (bellom) [inline src/components/BooksForm.js:8]: Modern React, we can always initialize our "state" without constructor.
example: `state = {firstName: ""}`
https://www.robinwieruch.de/react-state-without-constructor
**[Optional]**

File: src/components/BooksForm.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    super(props);
+    this.state = {
+      title: '',
+      category: CATEGORIES[0],
+    };
+
+    this.handleCategoryChange = this.handleCategoryChange.bind(this);
```
Comment (bellom) [inline src/components/BooksForm.js:15]: No need for `.bind(this)` when using arrow function.
_example:_ 
```
handleChange = (event) => {
    const {name, value} = event.target
    this.setState({ [name]: value })
}
```
https://stackoverflow.com/questions/52979915/why-we-dont-need-to-bind-the-arrow-function-in-react

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @maelfosso  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this milestone :+1: just a few issues left to work on............ :slightly_smiling_face: 

Below are part of the checklist for this milestone, kindly work on these few things listed below;
1. Kindly attend to my comments below;
2. We are missing the `Linter's` check on the `GitHub` page, see below;
image


**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by bellom** [comment]
> I just read the [Milestone 3](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896828/), and It like I've already implemented the feature asked.
> 
> How could I handle that, please?

Please how can you handle what?
I guess additional features you won't stop a TSE from reviewing, am just going to look into milestone 2 requirement alone.

File: src/components/BooksForm.js (hunk 7 above)
Comment (maelfosso) [reply src/components/BooksForm.js:15]: I try that but I got this error `Parsing error: Unexpected token =`

File: src/components/BooksForm.js (hunk 6 above)
Comment (maelfosso) [reply src/components/BooksForm.js:8]: I try that but I got the error `Parsing error: Unexpected token =`

Comment (bellom) [reply src/components/BooksForm.js:8]: You can go through the link I shared, i guess you're missing something.

**Pull request closed without merging**